source/Gamerules.cpp

```diff
@@ -46,6 +46,14 @@ void Gamerules::Load(const DataNode &node)
 			npcMaxMiningTime = max<int>(0, child.Value(1));
 		else if(key == "universal frugal threshold")
 			universalFrugalThreshold = min<double>(1., max<double>(0., child.Value(1)));
+		else if(key == "depreciation min")
+			depreciationMin = min<double>(1., max<double>(0., child.Value(1)));
+		else if(key == "depreciation daily")
+			depreciationDaily = min<double>(1., max<double>(0., child.Value(1)));
+		else if(key == "depreciation grace period")
+			depreciationGracePeriod = max<int>(0, child.Value(1));
+		else if(key == "depreciation max age")
+			depreciationMaxAge = max<int>(0, child.Value(1)) + depreciationGracePeriod;
```

Comment:
This addition shouldn't be done here.
A new grace period may be loaded after the last time a max age value is loaded, which will make the stored max age value incorrect.
Either do the addition in a `Gamerules::FinishLoading()` method (which you'll have to add), `Gamerules::DepreciationMaxAge()`, or `Depreciation.cpp <unnamed>::MaxAge()` (the namespace mmethod in Depreciation.cpp.)
I would prefer the last option.
```suggestion
		else if(key == "depreciation max age")
			depreciationMaxAge = max<int>(0, child.Value(1));
```